fvm/src/machine/default.rs

```diff
@@ -37,6 +37,9 @@ pub struct DefaultMachine<B, E> {
     state_tree: StateTree<BufferedBlockstore<B>>,
     /// Mapping of CIDs to builtin actor types.
     builtin_actors: Manifest,
+    /// Somewhat unique ID of the machine consisting of (epoch, randomness)
+    /// randomness is generated with `initial_state_root`
+    id: MachineId,
```

Comment:
I'd just use a string. The ID is opaque.

Reply:
so `id: String` and `Self { id: machine_id.to_string(), .. }` ?